test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java

```diff
@@ -95,25 +104,64 @@ protected void doRun() throws Exception {
                 }
                 if (stoppingThread.isAlive()) {
                     logger.warn("failed to stop node [{}]'s threads within [{}] millis. Stopping thread stack trace:\n {}"
-                        , disruptedNode, getStoppingTimeoutInMillis(), stackTrace(stoppingThread));
+                        , disruptedNode, getStoppingTimeoutInMillis(), stackTrace(stoppingThread.getStackTrace()));
                     stoppingThread.interrupt(); // best effort;
                     throw new RuntimeException("stopping node threads took too long");
                 }
+                // block detection checks if other threads are blocked waiting on an object that is held by one
+                // of the threads that was suspended
+                if (isBlockDetectionSupported()) {
+                    blockDetectionThread = new Thread(new AbstractRunnable() {
+                        @Override
+                        public void onFailure(Exception e) {
+                            if (e instanceof InterruptedException == false) {
+                                throw new AssertionError("unexpected exception in blockDetectionThread", e);
+                            }
+                        }
+
+                        @Override
+                        protected void doRun() throws Exception {
+                            while (Thread.currentThread().isInterrupted() == false) {
+                                ThreadInfo[] threadInfos = threadBean.dumpAllThreads(true, true);
+                                for (ThreadInfo threadInfo : threadInfos) {
+                                    if (threadInfo.getThreadName().contains("[" + disruptedNode + "]") == false &&
+                                        threadInfo.getLockOwnerName() != null &&
+                                        threadInfo.getLockOwnerName().contains("[" + disruptedNode + "]")) {
+
+                                        // find ThreadInfo object of the blocking thread (if available)
```

Comment:
I'm not sure. I would expect it to be always available but. I wanted to be extra safe here though so that when we find an occurrence of a blocking scenario (which might happen very rarely) we are informed with all information that we have (and that it's not screwed up by a NullPointerException).